apps/web/src/components/Shared/Unfollow.tsx

```diff
@@ -28,16 +27,20 @@ const Unfollow: FC<UnfollowProps> = ({ profile, showText = false, setFollowing }
   const currentProfile = useAppStore((state) => state.currentProfile);
   const [writeLoading, setWriteLoading] = useState(false);
   const { isLoading: signLoading, signTypedDataAsync } = useSignTypedData({ onError });
-  const { data: signer } = useSigner();
 
   const burnWithSig = async (signature: string, typedData: CreateBurnEip712TypedData) => {
     const { tokenId, deadline } = typedData.value;
     const { v, r, s } = splitSignature(signature);
     const sig = { v, r, s, deadline };
 
-    const followNftContract = new Contract(typedData.domain.verifyingContract, FollowNft, signer as Signer);
+    const config = await prepareWriteContract({
```

Comment:
Since you obtain the contract's address and parameters dynamically inside the function, you can't use usePrepareWriteContract because hooks inside a function can't be called